src/include/misc.h

```diff
@@ -257,15 +257,22 @@
 #endif
 
 #ifdef HAVE_TIMESTAMPS
-#define	WT_TIMESTAMP(x) (x)
-typedef uint8_t wt_timestamp_t[WT_TIMESTAMP_SIZE];
+#define	WT_TIMESTAMP_PTR(x) (&x)
```

Comment:
@keithbostic We touched on this code change last night when we talked. I wanted to say why I pushed @sulabhM towards the structure we have here. What I want to avoid is having to use

```
#if WT_TIMESATMP_SIZE == 8
#elif HAVE_TIMESTAMPS
#endif
```

For every place that deals with timestamps. What I really want to be able to do is to only care that the size is 8 in cases where it's performance critical. Sulabh suggested just casting the array, but I don't believe that is going to  guarantee well enough an atomic read/write, which is what we need.

It would be great if you can let us know your thoughts on how you'd approach this change.